wallet-test/src/test/scala/org/bitcoins/wallet/FundTransactionHandlingTest.scala

```diff
@@ -0,0 +1,107 @@
+
+    val wallet = fundedWallet.wallet
+    val fundedTxF = wallet.fundRawTransaction(
+      destinations = Vector(destination),
+      feeRate = feeRate)
+    for {
+      fundedTx <- fundedTxF
+    } yield {
+      assert(fundedTx.inputs.length == 1, s"We should only need one input to fund this tx")
+      assert(fundedTx.outputs.contains(destination))
+      assert(fundedTx.outputs.length == 2, s"We must have a single destination output and a change output")
+    }
+  }
+
+  it must "fund a transaction that requires all utxos in our wallet" in { fundedWallet: FundedWallet =>
+    val amt = Bitcoins(5.5)
+    val newDestination = destination.copy(value = amt)
+    val wallet = fundedWallet.wallet
+    val fundedTxF = wallet.fundRawTransaction(
+      destinations = Vector(newDestination),
+      feeRate = feeRate)
+    for {
+      fundedTx <- fundedTxF
+    } yield {
+      assert(fundedTx.inputs.length == 3, s"We should need 3 inputs to fund this tx")
+      assert(fundedTx.outputs.contains(newDestination))
+      assert(fundedTx.outputs.length == 2, s"We must have a 2 destination output and a change output")
+    }
+  }
+
+  it must "not care about the number of destinations" in { fundedWallet: FundedWallet =>
+    val destinations = Vector.fill(5)(destination)
+
+    val wallet = fundedWallet.wallet
+    val fundedTxF = wallet.fundRawTransaction(
+      destinations = destinations,
+      feeRate = feeRate)
+    for {
+      fundedTx <- fundedTxF
+    } yield {
+      assert(fundedTx.inputs.length == 1, s"We should only need one input to fund this tx")
+
+      destinations.foreach(d =>
+        assert(fundedTx.outputs.contains(d))
+      )
+      assert(fundedTx.outputs.length == 6, s"We must have a 6 destination output and a change output")
+    }
+
+  }
+
+  it must "fail to fund a raw transaction if we don't have enough money in our wallet" in { fundedWallet: FundedWallet =>
+    //our wallet should only have 6 bitcoin in it
+    val tooMuchMoney = Bitcoins(10)
+    val tooBigOutput = destination.copy(value = tooMuchMoney)
+    val wallet = fundedWallet.wallet
+    val fundedTxF = wallet.fundRawTransaction(
+      destinations = Vector(tooBigOutput),
+      feeRate = feeRate)
+
+    recoverToSucceededIf[RuntimeException] {
+      fundedTxF
+    }
+  }
+
+  it must "fail to fund a raw transaction if we have the _exact_ amount of money in the wallet because of the fee" in {fundedWallet: FundedWallet =>
+    //our wallet should only have 6 bitcoin in it
+    val tooMuchMoney = Bitcoins(6)
+    val tooBigOutput = destination.copy(value = tooMuchMoney)
+    val wallet = fundedWallet.wallet
+
+    //6 bitcoin destination + 1 sat/vbyte fee means we should
+    //not have enough money for this
+    val fundedTxF = wallet.fundRawTransaction(
+      destinations = Vector(tooBigOutput),
+      feeRate = feeRate)
+
+    recoverToSucceededIf[RuntimeException] {
```

Comment:
Can we give a more specific type than `RuntimeException` here?

Reply:
This is thrown inside of [CoinSelector](https://github.com/bitcoin-s/bitcoin-s/blob/82ef36e2a612175817638e4b0d0194c5301b990b/wallet/src/main/scala/org/bitcoins/wallet/api/CoinSelector.scala#L61), I'm open to changing it in the future but I don't want to it in this PR as the diff is pretty big.